python/graphstorm/model/edge_decoder.py

```diff
@@ -927,7 +935,15 @@ def forward(self, g, h):
                 rel_embedding = rel_embedding.repeat(1,dest_emb.shape[0]).T
                 scores_etype = calc_distmult_pos_score(src_emb, dest_emb, rel_embedding)
 
-                weight = _get_edge_weight(g, LP_DECODER_EDGE_WEIGHT, canonical_etype)
+                if e_h is not None and canonical_etype in e_h.keys():
+                    weight = e_h[canonical_etype]
+                    assert th.is_tensor(weight), \
+                        "The edge weight for Link prediction must be a torch tensor."
```

Comment:
it's possible that there are multiple edge features. I think the error message should say that this decoder doesn't support multiple edge data.

Reply:
For edge_weight, it should be a single edge feature